### ISSUE TITLE
buildx with network=none generates entitlement error

### ISSUE THREAD

**Issue by deitch** [docker/buildx#524]
If I try to run with a builder with `--network=none`, I get the following error:

```
error: failed to solve: rpc error: code = Unknown desc = granting entitlement network.host is not allowed by build daemon configuration
```

I think the issue is from [here](https://github.com/docker/buildx/blob/33e3ca524e2c90e4dcd3a6d63eefc9559ea8236e/build/build.go#L467-L474), but unsure:

```go
	switch opt.NetworkMode {
	case "host", "none":
		so.FrontendAttrs["force-network-mode"] = opt.NetworkMode
		so.AllowedEntitlements = append(so.AllowedEntitlements, entitlements.EntitlementNetworkHost)
	case "", "default":
	default:
		return nil, nil, errors.Errorf("network mode %q not supported by buildkit", opt.NetworkMode)
	}
```

I could see why that would be the case if I set `--network=host`, which is _elevated_ privileges, but why do I get the error with `--network=none`, which is _reduced_ privileges?

Is there a workaround I could use?

**Comment by tonistiigi** [comment]
You are correct. `network=none` does not actually require entitlement to be allowed. Fix in #531

**Comment by deitch** [comment]
Thanks for the quick response @tonistiigi .

I had to dig (found another issue to which you responded), but eventually was able to find that you needed to create the builder with `--buildkitd-flags '--allow-insecure-entitlement network.host'` to get it to work for `--network=host` or (until the related PR is merged in) `--network=none`. It would be helpful to have it in the docs, if at all possible.
